src/util/hash-list.h

```diff
@@ -131,19 +228,16 @@ template<class I, class T> class HashList {
 
   std::vector<HashBucket> buckets_;
 
-  Elem *freed_head_;  // head of list of currently freed elements. [ready for
-  // allocation]
+  // head of list of currently freed elements. [ready for allocation]
+  Elem *freed_head_;
 
-  std::vector<Elem*> allocated_;  // list of allocated blocks.
-
-  static const size_t allocate_block_size_ = 1024;  // Number of Elements to
-  // allocate in one block.  Must be largish so storing allocated_ doesn't
-  // become a problem.
-};
+  std::vector<Elem *> allocated_;  // list of allocated blocks.
 
+  // Number of Elements to allocate in one block.  Must be largish so storing
+  // allocated_ doesn't become a problem.
+  static const size_t allocate_block_size_ = 1024;
 
+  static const size_t boundary_tag_ = static_cast<size_t>(-1);
+};
 }  // end namespace kaldi
-
-#include "util/hash-list-inl.h"
```

Comment:
I don't think this is good practice. The rest of the kaldi codebase uses -inl.h files to provide implementations of templates.

Reply:
OK, I will do it that way.
But I think we don't need to separate every declaration and its implementation, especially it is inline method. Writing them together is more straightforward.